cross-chain/solana/programs/tbtc/src/lib.rs

```diff
@@ -0,0 +1,357 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    associated_token::AssociatedToken,
+    // mint,
+    token::{mint_to, TokenAccount, Mint as SplMint, MintTo as SplMintTo, Token as SplToken},
+};
+
+declare_id!("HksEtDgsXJV1BqcuhzbLRTmXp5gHgHJktieJCtQd3pG");
+
+#[program]
+pub mod tbtc {
+    use super::*;
+    
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let tbtc = &mut ctx.accounts.tbtc; 
+        tbtc.authority = ctx.accounts.authority.key();
```

Comment:
Minor issue, but this authority should be the same pubkey as the one who deployed the program. The upgrade authority (found in this program's program data, which is an account owned by `BpfLoaderUpgradeable`) will be this authority. Unlikely, but someone can race you to initialize the instruction after you deploy, setting himself as the authority.

Also a note about upgradeability: how do you plan on managing this? Multisig? You should set the upgrade authority to your multisig address in this instruction. Or it can be managed by an external program, where that external program's PDA can be the upgrade authority.

Reply:
I think that after the deployment on `mainnet` we want to set the upgrade authority as Threshold Council multisig.